### DISCUSSION TITLE
cornerstone, marker, and zeptoforth prompt

### DISCUSSION THREAD

**Discussion by robzed** [tabemann/zeptoforth#32] [General]
I'm not sure of the difference between cornerstone and marker - it appears that the only difference is that the word produced by cornerstone doesn't get erased by cornerstone - is this correct? But what are the expected uses of this, and how are these used in practice. 

An example: If I have three fragments (or files) at the top of each one can I write something that ensures the previous one is erased? By hand on the terminal, cornerstone seems much more convenient. But with marker I get an error on the line the first time (which I could put special case code in for). 

Also it appears I get the boot prompt after using restore-state or other new cornerstones? Is this correct?
  Welcome to zeptoforth
  Built for rp2040, version 0.57.0, on Thu Jan 12 07:13:58 PM CST 2023
  zeptoforth comes with ABSOLUTELY NO WARRANTY: for details type `license'

Also - I'd like to detect a Zeptoforth board for automation - is it possible to get Zeptoforth to tell me who it is? Is there an existing word that tells me the Forth name, or even product the prompt above?

**Comment by tabemann** [comment]
`cornerstone` and `marker` are as you describe them - a word created with `cornerstone` erases everything in flash up to but not including itself, while a word created with `marker` erases everything in flash up to and including itself. Note that erasing flash (except for flash used for storing blocks) is always followed by a reboot (and hence a welcome message) to avoid leaving the system in an undefined state.

As for having zeptoforth identify itself there is `welcome`, which actually displays the welcome message. It in turn calls `kernel-platform`, `kernel-version`, and `kernel-date`, which are user-generated at build time, which return the current platform, kernel version, and kernel build date.

**Reply by robzed** [reply]
I don’t see welcome in the documentation. This is helpful. 

Understand about the reboot after marker/cornerstone. 

I guess in terms of use *inside* files marker is easier than cornerstone because it would cause a double definition.

**Reply by tabemann** [reply]
I forgot to put `welcome` in the docs because, even though I did not put it in the `internal` wordlist, it really was not designed for user usage.

**Reply by robzed** [reply]
You never know what might be useful :-)

**Reply by robzed** [reply]
Thank you
